commit 5f123887b78a2e766338519501c94b26c963e18e
Author: Fernando Perez <Fernando.Perez@berkeley.edu>
Date:   Sat Apr 24 21:42:49 2010 -0700

    Set Control-L to clear the screen when readline supports it.
    
    Closes https://bugs.launchpad.net/ipython/+bug/435519
    
    Note that before we had C-L to list completions (which tab does anyway), so
    this is a minor change in behavior.  Users can get the previous behavior if
    they want with a simple config setting, but C-L is the standard 'screen
    clear' keybinding.

Index: ipython-0.10/IPython/Extensions/ipy_defaults.py
===================================================================
--- ipython-0.10.orig/IPython/Extensions/ipy_defaults.py
+++ ipython-0.10/IPython/Extensions/ipy_defaults.py
@@ -40,7 +40,7 @@ o.readline = 1
 
 rlopts = """\
 tab: complete
-"\C-l": possible-completions
+"\C-l": clear-screen
 set show-all-if-ambiguous on
 "\C-o": tab-insert
 "\M-i": "    "
Index: ipython-0.10/IPython/UserConfig/ipythonrc
===================================================================
--- ipython-0.10.orig/IPython/UserConfig/ipythonrc
+++ ipython-0.10/IPython/UserConfig/ipythonrc
@@ -394,10 +394,8 @@ wxversion 0
 readline_parse_and_bind tab: complete
 #readline_parse_and_bind tab: menu-complete
 
-# This binds Control-l to printing the list of all possible completions when
-# there is more than one (what 'complete' does when hitting TAB twice, or at
-# the first TAB if show-all-if-ambiguous is on)
-readline_parse_and_bind "\C-l": possible-completions
+# This binds Control-l to clear screen functionality
+readline_parse_and_bind "\C-l": clear-screen
 
 # This forces readline to automatically print the above list when tab
 # completion is set to 'complete'. You can still get this list manually by
